koocook_core/tests/test_models.py

```diff
@@ -0,0 +1,593 @@
+
+
+class TestTagLabelModel(djangotest.TestCase):
+    def test_init(self):
+        # TODO: add more cases
+        with self.subTest('common values'):
+            label = TagLabel.objects.create(name='cuisine')
+            Tag.objects.create(name='French', label=label)
+        with self.subTest('duplicates'):
+            with self.assertRaises(ValueError):
+                Tag.objects.create(name='French', label=label)
+            try:
+                Tag.objects.create(name='Chinese', label=label)
+            except Exception as e:
+                raise self.failureException(
+                    'unexpected exception raised') from e
+
+    def test_fields_settings(self):
+        tag = Tag()
+        with self.subTest(field='name', attr='max_length'):
+            self.assertEqual(tag._meta.get_field('name').max_length, 50)
+        with self.subTest(field='label', attr='null'):
+            self.assertTrue(tag._meta.get_field('label').null)
+        with self.subTest(field='label', attr='blank'):
+            self.assertTrue(tag._meta.get_field('label').blank)
+
+
+class TestTagModel(djangotest.TestCase):
+    def test_init(self):
+        # TODO: add more cases
+        with self.subTest('common values'):
+            TagLabel.objects.create(name='cuisine')
+        with self.subTest('duplicates'):
+            with self.assertRaises(ValueError):
+                TagLabel.objects.create(name='cuisine')
+
+    def test_fields_settings(self):
+        label = TagLabel()
+        with self.subTest(field='name', attr='max_length'):
+            self.assertEqual(label._meta.get_field('name').max_length, 50)
+
+
+class TestRecipeIngredientModel(djangotest.TestCase):
+    def test_init(self):
+        # TODO: test common values
+        quantity = Quantity(Fraction(1, 2), 'tbsp')
+        mi = MetaIngredient.objects.create(name='')
+        recipe = Recipe.objects.create(name='')
+        try:
+            RecipeIngredient.objects.create(quantity=quantity,
+                                            meta=mi,
+                                            recipe=recipe)
+        except Exception as e:
+            raise self.failureException(
+                'unexpected exception raised') from e
+
+    def test_fields_settings(self):
+        ri = RecipeIngredient()
+        with self.subTest(field='substitute_set', attr='blank'):
+            self.assertTrue(ri._meta.get_field('substitute_set').blank)
+        for field, attr in (
+            ('quantity', 'null'),
+            ('quantity', 'blank'),
+            ('meta', 'null'),
+            ('meta', 'blank'),
+            ('recipe', 'null'),
+            ('recipe', 'blank'),
+        ):
+            with self.subTest(field=field, attr=attr):
+                self.assertFalse(getattr(ri._meta.get_field(field), attr))
+
+    def test_as_dict(self):
+        pass
+
+    def test_as_json(self):
+        pass
+
+    def test_nutrition(self):
+        pass
```

Comment:
This seems like a frame. When will you work on this?
Or it was waiting for the actual implementation?

Reply:
Waiting for the implementation details. IDK the details so I just put it there as a reminder to make the test

> When will you work on this?

No I won't work on these tests